### PR TITLE
Feat: Convert modmail to forum channel, other changes

### PR DESCRIPTION
**Pull request by WhoIsConch**
Converted modmail system to use a forum channel. Cleaned up the code.

## Summary
This pull request changes the ModMail system to use forum channels instead of text channels with threads.

## Checklist

- [x] If code changes were made then they have been tested.
- [x] This PR fixes an issue.

### CODE REVIEW

**Comment by WhoIsConch** [comment]
Oh yeah I also bumped the version of Dpy to latest

File: cogs/modmail.py [hunk 1]
```diff
@@ -1,55 +1,90 @@
...
 
     @commands.Cog.listener()
     async def on_message(self, message: discord.Message):
-        if (
-            not self.channel
-        ):  # I was unsure if getting the channel in the __init__ is wise
-            self.channel = self.bot.get_channel(MODMAIL_CHANNEL_ID)
+        if not self.channel:  
+            self.channel: discord.ForumChannel = self.bot.get_channel(MODMAIL_CHANNEL_ID)
+            print(self.channel.available_tags)
```
Comment (iayushanand) [inline cogs/modmail.py]: seems like useless print message

Comment (WhoIsConch) [reply cogs/modmail.py]: True. Will update

**Review by iayushanand — APPROVED** [review]
lgtm

**Review by andrewthederp — COMMENTED** [review]
Why are you fetching the webhook like that instead of `webhook = thread.parent.webhooks[0]`

**Comment by WhoIsConch** [comment]
> Why are you fetching the webhook like that instead of `webhook = thread.parent.webhooks[0]`

Good idea. I'll change that up quick.

**Review by andrewthederp — APPROVED** [review]

**Pull request merged**